### PR TITLE
feature: update lifecycle for S3C changes

### PR DESCRIPTION
**Pull request by RachedBenMustapha**
Updates the lifecycle design doc to account for new changes:
- Vault involvement
- removal of bucket list from zookeeper

### CODE REVIEW

File: docs/design/lifecycle-expiration.md [hunk 1]
```diff
@@ -39,52 +39,88 @@ other.
 Here's an example workflow that expires version *v1* of object *obj*
 from bucket *foobucket*:
 
-```
- __________________
-/                  \
```
Comment (jonathan-gramain) [inline docs/design/lifecycle-expiration.md:44]: RIP my beautiful ASCII art :cry:

**Review by jonathan-gramain — APPROVED** [review]

File: docs/design/lifecycle-expiration.md [hunk 2]
```diff
@@ -39,52 +39,88 @@ other.
...
-    |                                  |                            |  |
-    |                                  \----------------------------/  |
-    | DELETE foobucket/obj@v1            process:foobucket@(marker+n)  |
-    |                     __________________                           |
-    \--------------------/                  \<-------------------------/
-                         |    Lifecycle     |  expire:foobucket/obj@v1
-                         | Object Processor |
-                         \__________________/
-
+```mermaid
```
Comment (vrancurel) [inline docs/design/lifecycle-expiration.md:42]: The Mermaid is not rendered in my view.

Comment (miniscruff) [reply docs/design/lifecycle-expiration.md:42]: Mermaid JS requires some javascript so it won't render in githubs preview, you will have to run it locally or wait for it to be merged to see the diagram. ( I described the how to in the readme, its not too bad )

Comment (jonathan-gramain) [reply docs/design/lifecycle-expiration.md:42]: might or not work for you: https://mermaid-js.github.io/mermaid-live-editor/view/#eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gIGF1dG9udW1iZXJcbiAgcGFydGljaXBhbnQgY2xpZW50IGFzIFMzIGNsaWVudFxuICBwYXJ0aWNpcGFudCBzM2MgYXMgUzMgQ29ubmVjdG9yXG4gIHBhcnRpY2lwYW50IHYgYXMgVmF1bHRcbiAgcGFydGljaXBhbnQgbWQgYXMgTWV0YWRhdGFcbiAgcGFydGljaXBhbnQgYnAgYXMgQnVja2V0IFByb2Nlc3NvclxuICBwYXJ0aWNpcGFudCBjIGFzIENvbmR1Y3RvclxuICBwYXJ0aWNpcGFudCBrIGFzIEthZmthXG4gIHBhcnRpY2lwYW50IHogYXMgWm9va2VlcGVyXG4gIHBhcnRpY2lwYW50IG9wIGFzIE9iamVjdCBQcm9jZXNzb3JcbiAgY2xpZW50LT4-czNjOiBQVVQgL2Zvb2J1Y2tldD9saWZlY3ljbGVcbiAgczNjLT4-djogQXV0aGVudGljYXRlIGFuZCBhdXRob3JpemVcbiAgczNjLT4-bWQ6IFdyaXRlIGJ1Y2tldCBsaWZlY3ljbGUgcG9saWN5IGluIG1ldGFkYXRhIGZvciBmb29idWNrZXRcbiAgbG9vcCBMaWZlY3ljbGUgQmF0Y2hcbiAgICBjLT4-YzogV2FrZSB1cCBhdCBjb25maWd1cmVkIHNjaGVkdWxlXG4gICAgYy0-Pno6IENoZWNrIHB1Ymxpc2hlZCBidWNrZXQvb2JqZWN0IHRvcGljIG9mZnNldHNcbiAgICBvcHQgRWFybHkgY2hlY2tcbiAgICAgIGMtPj5jOiBCYWlsIGlmIHByZXZpb3VzIGJhdGNoIHN0aWxsIGluIHByb2dyZXNzXG4gICAgZW5kXG4gICAgYy0-Pm1kOiBMaXN0IChBY2NvdW50IENhbm9uaWNhbCBJRHMsIEJ1Y2tldCkgcGFpcnNcbiAgICBjLT4-djogR2V0IEFjY291bnQgSURzIGZvciBBY2NvdW50IENhbm9uaWNhbCBJRHNcbiAgICBjLT4-azogUHVzaCBtZXNzYWdlOiBwcm9jZXNzKGZvb2J1Y2tldCBAIG51bGxtYXJrZXIsIGZvb0FjY291bnRJZClcbiAgZW5kXG4gIGxvb3AgQnVja2V0IFByb2Nlc3NvclxuICAgIGJwLT4-azogUHVsbCBwZW5kaW5nIG1lc3NhZ2UocylcbiAgICBhbHQgTWVzc2FnZSBUeXBlIHByb2Nlc3MoYnVja2V0IEAgbWFya2VyLCBhY2NvdW50SWQpXG4gICAgICBicC0-PnY6IEFzc3VtZVJvbGUoYXJuRm9yQWNjb3VudElkKHNlZWRlZExpZmVjeWNsZVJvbGUpKVxuICAgICAgYnAtPj5zM2M6IENoZWNrIGlmIGJ1Y2tldCBsaWZlY3ljbGVkXG4gICAgICBzM2MtPj52OiBBdXRoZW50aWNhdGUgYW5kIGF1dGhvcml6ZVxuICAgICAgczNjLT4-bWQ6IEdFVCBidWNrZXQgYXR0cmlidXRlc1xuICAgICAgYnAtPj5zM2M6IExpc3Qgb2JqZWN0ICYgTVBVcyBuS2V5cyBzdGFydGluZyBhdCBtYXJrZXIgZnJvbSBmb29idWNrZXRcbiAgICAgIHMzYy0-PnY6IEF1dGhlbnRpY2F0ZSBhbmQgYXV0aG9yaXplXG4gICAgICBzM2MtPj5tZDogR0VUIGJ1Y2tldCBsaXN0aW5nXG4gICAgICBhbHQgRW5kIG9mIGxpc3Rpbmcgbm90IHJlYWNoZWRcbiAgICAgICAgYnAtPj5rOiBQdXNoIG1lc3NhZ2UgcHJvY2Vzcyhmb29idWNrZXQgQCBtYXJrZXIgKyBuS2V5cywgYWNjb3VudElkKVxuICAgICAgZW5kXG4gICAgICBsb29wIEVhY2ggb2JqZWN0QHZlcnNpb24gaW4gbGlzdGluZ1xuICAgICAgICBicC0-PnMzYzogR2V0IG1ldGFkYXRhIHRvIGV2YWx1YXRlIHZzIGJ1Y2tldCBsaWZlY3ljbGUgY29uZmlndXJhdGlvblxuICAgICAgICBzM2MtPj52OiBBdXRoZW50aWNhdGUgYW5kIGF1dGhvcml6ZVxuICAgICAgICBzM2MtPj5tZDogR0VUIG9iakB2MSBtZFxuICAgICAgICBvcHQgSWYgcnVsZSBtYXRjaFxuICAgICAgICAgIGJwLT4-azogUHVzaCBtZXNzYWdlIGV4cGlyZShmb29idWNrZXQvb2JqQHYxLCBhY2NvdW50SWQpXG4gICAgICAgIGVuZFxuICAgICAgZW5kXG4gICAgZW5kXG4gICAgbG9vcCBFdmVyeSA2MHNcbiAgICAgIGJwLT4-ejogUHVibGlzaCB0b3BpYyBvZmZzZXRzXG4gICAgZW5kXG4gIGVuZFxuICBsb29wIE9iamVjdCBQcm9jZXNzb3JcbiAgICBvcC0-Pms6IFB1bGwgcGVuZGluZyBtZXNzYWdlKHMpXG4gICAgYWx0IE1lc3NhZ2UgVHlwZSBleHBpcmUoZm9vYnVja2V0L29iakB2MSwgYWNjb3VudElkKVxuICAgICAgb3AtPj52OiBBc3N1bWVSb2xlKGFybkZvckFjY291bnRJZChzZWVkZWRMaWZlY3ljbGVSb2xlKSlcbiAgICAgIG9wLT4-czNjOiBDaGVjayBmb29idWNrZXQvb2JqQHYxIGxhc3QgbW9kaWZpZWQgZGF0ZVxuICAgICAgczNjLT4-djogQXV0aGVudGljYXRlIGFuZCBhdXRob3JpemVcbiAgICAgIHMzYy0-Pm1kOiBHRVQgb2JqQHYxIG1kXG4gICAgICBvcHQgSWYgbm9uLXZlcnNpb25lZCBidWNrZXRcbiAgICAgICAgb3AtPj5vcDogQmFpbCBpZiBvYmplY3QgaGFzIGJlZW4gbW9kaWZpZWQgc2luY2UgcXVldWVkXG4gICAgICBlbmRcbiAgICAgIG9wLT4-czNjOiBERUxFVEUgb3IgQWJvcnRNdWx0aXBhcnRVcGxvYWQgb2JqQHYxXG4gICAgICBzM2MtPj52OiBBdXRoZW50aWNhdGUgYW5kIGF1dGhvcml6ZVxuICAgICAgczNjLT4-bWQ6IERFTEVURVxuICAgIGVuZFxuICAgIGxvb3AgRXZlcnkgNjBzXG4gICAgICBvcC0-Pno6IFB1Ymxpc2ggdG9waWMgb2Zmc2V0c1xuICAgIGVuZFxuICBlbmQiLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjp0cnVlLCJhdXRvU3luYyI6ZmFsc2UsInVwZGF0ZURpYWdyYW0iOmZhbHNlfQ

File: docs/design/lifecycle-expiration.md (hunk 1 above)
Comment (RachedBenMustapha) [reply docs/design/lifecycle-expiration.md:44]: Really sorry, it was beautiful indeed, tried to keep it but got hairy.

**Review by nicolas2bert — APPROVED** [review]

**Pull request merged**